add solution

vk-workshop opened the pull request:
[DEMO LINK](https://vk-workshop.github.io/react_dynamic-list-of-todos/)

etojeDenys commented on src/components/TodoModal/TodoModal.tsx:
```diff
@@ -1,49 +1,82 @@
+    return null;
+  }
+
+  const { title, id, completed } = selectTodo || {};
+  const { email, name } = user || {};
```
```suggestion
  const { title, id, completed } = selectTodo;
  const { email, name } = user;
```

etojeDenys commented on src/components/TodoModal/TodoModal.tsx:
```diff
@@ -1,49 +1,82 @@
+    isLoadingModal,
+  } = useContext(TodosContext);
+
+  if (!selectTodo && !user) {
```
```suggestion
  if (!selectTodo || !user) {
```

etojeDenys commented on src/components/State/State.tsx:
```diff
@@ -0,0 +1,105 @@
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [selectOption, setSelectOption] = useState('');
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
```
it's can be the simple var instead of the state

vk-workshop commented:
- changed logic !selectTodo && !user to !selectTodo || !user and lost Loader in TodoModal component
- changed var instead of the state where was possible

Andriy-Shymkiv reviewed (changes requested):
I can't insert a space in search input, please fix it

vk-workshop commented:
- deleted redundant code
- changed state to `filteredTodos` function
